src/Adapter/PhpredisAdapter.php

```diff
@@ -40,13 +29,16 @@ public function __construct(array $config = [], bool $lazy = true)
 
     protected function getClient(): Redis
     {
-        if (!$this->client instanceof Redis) {
-            $this->createClient($this->config);
+        if (!$this->client) {
```

Comment:
```suggestion
        if (!isset($this->client)) {
```
Would probably fix the error reported by @damian-af

Reply:
Just checked and yes, that seems to work.

Reply:
That seems to be a failure of phpstan to understand that `createClient` causes a side-effect of `$this->client` being defined. It's not so good for people too. I'm going to make `createClient` return a pointer to the instance just created so it is clearer.